filters.go

```diff
@@ -157,3 +159,38 @@ func (p *Pipe) First(lines int) *Pipe {
 	}
 	return Echo(output.String())
 }
+
+// Freq reads from the pipe, and returns a new pipe containing only unique lines
+// from the input, prefixed with a frequency count, in descending numerical
+// order (most frequent lines first). Lines with equal frequency will be sorted
+// alphabetically. If there is an error reading the pipe, the pipe's error
+// status is also set.
+func (p *Pipe) Freq() *Pipe {
+	if p == nil || p.Error() != nil {
+		return p
+	}
+	var freq = map[string]int{}
+	p.EachLine(func(line string, out *strings.Builder) {
+		freq[line]++
+	})
+	type frequency struct {
```

Comment:
We could check the len of the map and short cirquit the rest, if we have 0 elements in the map.

Reply:
Yes, that's a good suggestion, but there's not actually a lot to short-circuit. The `for` loop will execute zero times, the `sort.Slice` will be a no-op, and then there'll be no output!